File: src/abc288/c.py
import sys
sys.setrecursionlimit(10**6)

def dfs(c):
    vis[c] = 1
    for i in g[c]:
        if vis[i]==0:
            dfs(i)
    return


# main
n,m = map(int, input().split())

g = [[] for _ in range(n)]
for _ in range(m):
    u,v = map(int, input().split())
    g[u-1].append(v-1)
    g[v-1].append(u-1)

vis = [0] * n
cnt = 0
for i in range(n):
    if vis[i] == 0:
        dfs(i)
        cnt += 1

print(m - (n-cnt))

File: src/abc288/c.rs
use proconio::input;
// use std::collections::HashMap;

fn dfs(g: &Vec<Vec<usize>>, vis: &mut Vec<bool>, c: usize) {
    vis[c] = true;
    for &i in g[c].iter() {
        if vis[i] == false {
            dfs(g, vis, i)
        }
    }
}

fn main() {
    input! {
        n: usize,
        m: usize,
        ab: [(usize, usize); m],
    }

    let mut g: Vec<Vec<usize>> = vec![vec![]; n];

    for i in 0..m {
        g[ab[i].0 - 1].push(ab[i].1 - 1);
        g[ab[i].1 - 1].push(ab[i].0 - 1);
    }

    let mut vis: Vec<bool> = vec![false; n];
    let mut cnt: u32 = 0;
    for i in 0..n {
        if vis[i] == false {
            dfs(&g, &mut vis, i);
            cnt += 1;
        }
    }

    println!("{}", m as u32 - (n as u32 - cnt));
}